lib/puppet/provider/service/init.rb

```diff
@@ -11,6 +11,8 @@ def self.defpath
       "/sbin/init.d"
     when "Archlinux"
       "/etc/rc.d"
+    when "AIX"
+      "etc/rc.d/init.d"
```

Comment:
This is missing the leading `/`, which causes it to fail to find the path to `init.d` in AIX 5.3. I'm confused as to why it worked in 7.1 - maybe your local edit there had the `/`?